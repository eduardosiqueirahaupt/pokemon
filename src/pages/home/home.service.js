import axios from "axios";
const initialUrl = 'https://pokeapi.co/api/v2/pokemon/?offset=0&limit=20'
const byIdPokemon = 'https://pokeapi.co/api/v2/pokemon/'
export const allPokemonUrl = 'https://pokeapi.co/api/v2/pokemon?limit=100000&offset=0'
export const fetchData = (url = initialUrl) => (axios.get(url).then(x => x.data))
export const fetchAllById = (ids) => {
    return axios.all(ids.map((id) => axios.get(byIdPokemon + id).then(x => x.data)))
}