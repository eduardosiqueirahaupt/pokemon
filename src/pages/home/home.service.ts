import { IFetchPokemons } from ".";
import { useFetch } from "../../hooks/useFetch";
export interface IFetch {
  data: IFetchPokemons;
}
export const usePokemonsGetData = (): IFetch =>
  useFetch("https://pokeapi.co/api/v2/pokemon/?offset=0&limit=20");
