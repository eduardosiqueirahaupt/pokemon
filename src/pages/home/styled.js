import styled from 'styled-components'

export const Wrapper = styled.div`
  height: calc(100% - 50px);
  position: relative;
`

export const Content = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
`

export const WrapperCards = styled.div`
  width: 100%;
`

export const ContentCards = styled.div`
  display: grid;
  grid-template-columns: repeat(auto-fill, 150px);
  grid-row-gap: 12px;
  grid-column-gap: 28px;
  justify-content: center;
`