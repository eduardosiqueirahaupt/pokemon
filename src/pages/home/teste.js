// import { Snackbar } from "@mui/material";
// import { useCallback, useEffect, useState } from "react";
// import Cards from "../../components/cards";
// import CardsFavorites from "../../components/cardsFavorites";
// import Header from "../../components/header-cards";
// import { fetchData } from "./home.service";
// import * as S from "./styled";

// interface IHome {
//   isFavorites: boolean;
// }

// interface Ipokemon {
//   name: string;
// }

// const initialValueFeedback = {
//   isOpen: false,
//   message: "",
// };

// function Home({ isFavorites }: IHome) {
//   const [search, setSearch] = useState("");
//   const [pokeData, setPokeData] = useState<Ipokemon[]>([]);
//   const [feedback, setFeedback] = useState(initialValueFeedback);

//   const handleFeedback = (message: string) => {
//     setFeedback({
//       isOpen: true,
//       message: message,
//     });
//   };

//   const handleSearch = (e: React.ChangeEvent<HTMLInputElement>) => {
//     setSearch(e.target.value);
//   };

//   const getData = useCallback(async () => {
//     const _data = await fetchData();
//     setPokeData(_data.results);
//   }, []);

//   const filterPokemons = (): Ipokemon[] =>
//     pokeData.filter((x) => x.name.includes(search.toLowerCase()));

//   useEffect(() => {
//     getData();
//   }, []);

//   return (
//     pokeData && (
//       <S.Wrapper>
//         <Header handleSearch={handleSearch} search={search} />
//         <S.Content>
//           <S.WrapperCards>
//             <S.ContentCards>
//               {isFavorites ? (
//                 <CardsFavorites
//                   data={filterPokemons()}
//                   handleFeedback={handleFeedback}
//                 />
//               ) : (
//                 <Cards
//                   data={filterPokemons()}
//                   handleFeedback={handleFeedback}
//                 />
//               )}
//             </S.ContentCards>
//           </S.WrapperCards>
//         </S.Content>
//         <Snackbar
//           open={feedback.isOpen}
//           autoHideDuration={1000}
//           onClose={() => setFeedback(initialValueFeedback)}
//           message={feedback.message}
//         />
//       </S.Wrapper>
//     )
//   );
// }

// export default Home;

export default () => {
  return <div>
      testes
  </div>;
};
