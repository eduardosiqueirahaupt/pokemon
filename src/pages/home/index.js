import { Snackbar } from '@mui/material';
import { useCallback, useEffect, useState } from 'react';
import Cards from '../../components/cards';
import Header from '../../components/header-cards';
import Pagination from '../../components/pagination';
import Wrapper from '../../components/wrapper';
import { fetchData } from './home.service';

const initialValueFeedback = {
  isOpen: false,
  message: ''
}

const Home = () => {
  const [search, setSearch] = useState('');
  const [pokeData, setPokeData] = useState([]);
  const [feedback, setFeedback] = useState(initialValueFeedback)
  const [pagination, setPagination] = useState({
    next: '',
    previous: ''
  })

  const handleFeedback = (message) => {
    setFeedback({
      isOpen: true,
      message: message
    })
  }

  const handleSearch = (e) => {
    setSearch(e.target.value)
  }

  const filterPokemons = () => pokeData.
    filter(x => x.name.includes(search.toLowerCase()))

  const getData = useCallback(async (url) => {
    const _data = await fetchData(url);
    setPokeData(_data.results)
    setPagination({
      next: _data.next,
      previous: _data.previous
    })
  }, [])

  useEffect(() => {
    getData()
  }, [getData])

  return (
    <>
      <Header
        handleSearch={handleSearch}
        search={search}
      />
      <Wrapper>
        <Cards
          data={filterPokemons()}
          handleFeedback={handleFeedback}
        />
      </Wrapper>
      <Pagination
        previous={() => getData(pagination.previous)}
        next={() => getData(pagination.next)}
        pagination={pagination}
      />
      <Snackbar
        open={feedback.isOpen}
        autoHideDuration={1000}
        onClose={() => setFeedback(initialValueFeedback)}
        message={feedback.message}
      />
    </>
  )
}

export default Home;