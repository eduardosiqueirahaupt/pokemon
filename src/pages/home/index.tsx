import { useState } from "react";
import Cards from "../../components/cards";
import Header from "../../components/header-cards";
import { usePokemonsGetData } from "./home.service";
import * as S from "./styled";

interface Ipokemon {
  name: string;
}

export interface IFetchPokemons {
  results: Ipokemon[] | undefined;
}

function Home() {
  const [search, setSearch] = useState("");
  const { data } = usePokemonsGetData();

  const handleSearch = (e: React.ChangeEvent<HTMLInputElement>) => {
    setSearch(e.target.value);
  };

  const filterPokemons = () =>
    data?.results?.filter((x) => x.name.includes(search.toLowerCase()));

  if (!data?.results) {
    return <span>Carregando....</span>;
  }

  return (
    <S.Wrapper>
      <Header handleSearch={handleSearch} search={search} />
      <S.Content>
        <S.WrapperCards>
          <S.ContentCards>
            <Cards data={filterPokemons()} />
          </S.ContentCards>
        </S.WrapperCards>
      </S.Content>
    </S.Wrapper>
  );
}

export default Home;
