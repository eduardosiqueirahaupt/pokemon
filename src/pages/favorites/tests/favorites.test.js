import { cleanup, fireEvent, render, screen, waitFor } from '@testing-library/react'
import Favorites from '..'
import * as service from '../../home/home.service'

const renderComponent = () => (render(<Favorites />))


describe("<Favorites/>", () => {

  beforeAll(() => {
    localStorage.setItem("FavPokemons", JSON.stringify(['1', '2', '3', '4', '5']))
  });

  beforeEach(() => {
    jest.spyOn(service, 'fetchAllById').mockImplementation(jest.fn(() => Promise.resolve(getFavPokemons)));
    cleanup();
  })

  it("Deve renderizar com sucesso favoritos", async () => {
    const { container } = renderComponent();
    await waitFor(() => screen.findByText(/bulbasaur/))
    expect(container).toMatchSnapshot();
  });


  it("Deve remover dos favoritos e nÃ£o aparecer mais em tela", async () => {
    renderComponent();
    await waitFor(() => screen.findByText(/bulbasaur/));

    const favIconBulbasaur = screen.getByTestId('fav-icon-bulbasaur');
    fireEvent.click(favIconBulbasaur)

    expect(await waitFor(() => screen.findByText(/Pokemon Removido dos Favoritos/))).toBeDefined();
  });
})


const getFavPokemons = [
  { "name": "bulbasaur", "id": "1" },
  { "name": "ivysaur", "id": "2" },
  { "name": "venusaur", "id": "3" }
]