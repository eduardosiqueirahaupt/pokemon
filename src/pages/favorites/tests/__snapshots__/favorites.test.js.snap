// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Favorites/> Deve renderizar com sucesso favoritos 1`] = `
<div>
  <div
    class="sc-eCYdqJ jZqxvB"
  >
    <div
      class="sc-jSMfEi gOcqoV"
    >
      <div
        class="sc-gKXOVf bVVSER"
      >
        <div
          class="sc-iBkjds bRnbqh"
        >
          <div
            class="sc-bczRLJ hxuUtm"
          >
            <div
              class="sc-gsnTZi fUNCkM"
            >
              bulbasaur
            </div>
            <div
              class="sc-dkzDqf hFZDNz"
            >
              <img
                alt="poketeste"
                height="100%"
                src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/1.svg"
                width="90%"
              />
            </div>
            <img
              alt="fav-icon"
              class="sc-hKMtZM bvsQcw"
              data-testid="fav-icon-bulbasaur"
              src="favorito.png"
            />
          </div>
          <div
            class="sc-bczRLJ hxuUtm"
          >
            <div
              class="sc-gsnTZi fUNCkM"
            >
              ivysaur
            </div>
            <div
              class="sc-dkzDqf hFZDNz"
            >
              <img
                alt="poketeste"
                height="100%"
                src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/2.svg"
                width="90%"
              />
            </div>
            <img
              alt="fav-icon"
              class="sc-hKMtZM bvsQcw"
              data-testid="fav-icon-ivysaur"
              src="favorito.png"
            />
          </div>
          <div
            class="sc-bczRLJ hxuUtm"
          >
            <div
              class="sc-gsnTZi fUNCkM"
            >
              venusaur
            </div>
            <div
              class="sc-dkzDqf hFZDNz"
            >
              <img
                alt="poketeste"
                height="100%"
                src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/3.svg"
                width="90%"
              />
            </div>
            <img
              alt="fav-icon"
              class="sc-hKMtZM bvsQcw"
              data-testid="fav-icon-venusaur"
              src="favorito.png"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
